addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -0,0 +1,426 @@
+        def group_tax_retention(tax_values):
+            return {'l10n_eg_eta_code': tax_values['tax_id'].l10n_eg_eta_code.split('_')[0]}
+
+        date_string = invoice.invoice_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+        edi_values = invoice._prepare_edi_vals_to_export()
+        grouped_taxes = invoice._prepare_edi_tax_details(
+            grouping_key_generator=group_tax_retention)['tax_details'].values()
+        eta_invoice = {
+            'issuer': self._l10n_eg_eta_prepare_address_data(invoice.journal_id.l10n_eg_branch_id, issuer=True,),
+            'receiver': self._l10n_eg_eta_prepare_address_data(invoice.partner_id, invoice_amount=invoice.amount_total),
+            'documentType': 'i' if invoice.move_type == 'out_invoice' else 'c' if invoice.move_type == 'out_refund' else 'd' if invoice.move_type == 'in_refund' else '',
+            'documentTypeVersion': '1.0',
+            'dateTimeIssued': date_string,
+            'taxpayerActivityCode': invoice.journal_id.l10n_eg_branch_id.l10n_eg_activity_type_id.code,
+            'internalID': invoice.name,
+        }
+        if invoice.move_type in ['out_refund', 'in_refund']:
+            eta_invoice.update({
+                'references': [invoice.reversed_entry_id.l10n_eg_uuid] if invoice.reversed_entry_id.l10n_eg_uuid else []
+            })
+        eta_invoice.update({
+            'invoiceLines': self._l10n_eg_eta_prepare_invoice_lines_data(invoice),
+            'totalDiscountAmount': invoice._get_amount_main_currency(edi_values['total_price_discount']),
+            'totalSalesAmount': invoice._get_amount_main_currency(edi_values['total_price_subtotal_before_discount']),
+            'netAmount': invoice._get_amount_main_currency(invoice.amount_untaxed),
+            'taxTotals': [{
+                'taxType': tax['l10n_eg_eta_code'].split('_')[0].upper(),
+                'amount': abs(invoice._get_amount_main_currency(tax['tax_amount_currency'])),
+            } for tax in grouped_taxes],
+            'totalAmount': invoice._get_amount_main_currency(invoice.amount_total),
+            'extraDiscountAmount': 0.0,
+            'totalItemsDiscountAmount': 0.0,
+        })
+        return eta_invoice
+
+    def _l10n_eg_eta_prepare_invoice_lines_data(self, invoice):
+        lines = []
+        for line in invoice.invoice_line_ids.filtered(lambda x: x.display_type not in ('line_section', 'line_note')):
+            discount = (line.discount / 100.0) * line.quantity * line.price_unit
+            lines.append({
+                'description': line.name,
+                'itemType': line.product_id.barcode.startswith('EG') and 'EGS' or 'GS1',
+                'itemCode': line.product_id.barcode,
+                'unitType': line.product_uom_id.l10n_eg_unit_code_id.code,
+                'quantity': line.quantity,
+                'internalCode': line.product_id.default_code or '',
+                'salesTotal': invoice._get_amount_main_currency(line.quantity*line.price_unit),
```

Comment:
```suggestion
                'salesTotal': invoice._get_amount_main_currency(line.quantity * line.price_unit),
```
price_unit could be price included, while you need it excluded
```suggestion
                'salesTotal': abs(line.balance / (1 - (line.discount / 100))) / line.quantity),
``` (would need extra rounding)